extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -438,14 +438,14 @@ protected static OIDCException toOidcException(Throwable cause, String authServe
     protected static Uni<OidcProvider> createOidcProvider(OidcTenantConfig oidcConfig, Vertx vertx,
             OidcTlsSupport tlsSupport) {
         return createOidcClientUni(oidcConfig, vertx, tlsSupport)
-                .flatMap(new Function<OidcProviderClient, Uni<? extends OidcProvider>>() {
+                .flatMap(new Function<>() {
                     @Override
                     public Uni<OidcProvider> apply(OidcProviderClient client) {
                         if (oidcConfig.jwks.resolveEarly
                                 && client.getMetadata().getJsonWebKeySetUri() != null
                                 && !oidcConfig.token.requireJwtIntrospectionOnly) {
                             return getJsonWebSetUni(client, oidcConfig).onItem()
-                                    .transform(new Function<JsonWebKeySet, OidcProvider>() {
```

Comment:
I'm not sure this style is better, even though it is less verbose, though I can see IDE can generate a method correctly without these template params. 
Let me ask @geoand, @michalvavrik, what should we prefer, for ex, here, are we happy with `new Function<>` only, as far as the readability is concerned ?

Reply:
and if you have comments about other suggested updates, please comment

Reply:
looks like new Function<> is preferred, so let's keep these changes on, let me resolve, but you are all welcome to comment

Reply:
I don't have any preference here, but maybe using a lambda instead

Reply:
> I don't have any preference here, but maybe using a lambda instead

-1